### ISSUE TITLE
About page: Clicking Save and Delete buttons gives error in terminal

### ISSUE THREAD

**Issue by tas12** [MindAidPrototype/MindAidAdmin#77]
Error message:

Debug: internal, implementation, error
    TypeError: listener must be a function
    at addListener (events.js:210:11) ...

Functionality works but we get the above error in the terminal.

**Comment by shouston3** [comment]
Right, I've had a look through and I currently can't work out what is going on. I've got a vague feeling it may be something to do with the call back and mapping through the array.
The temporary fix is going to be to not close the db in the editAbout.js file (unless anyone feels like giving it a go before I get back, path to take: write tests for edit and delete functions and  rewrite edit and delete functions (without fetching all the data), then it will be much easier to debug and we'll probably solve it from there)
I think we should rewrite the editData function to use the [updateOne method](https://www.npmjs.com/package/mongodb), and change the params needed for changing data:
currently they are: index of element and new state of element
they should be: old element and new element (seems more sensible for the updateOne method)
Also, I'm confused by what the `if (index === 'new')` statement is for.
It would be much easier to understand everything about the function with tests

**Comment by tas12** [comment]
Thanks Sam, I'm happy with the temporary fix. Have you done this already on a local branch? 

I've opened a new issue #80 to refactor the editData and deleteData functions. I'll take this on if that's okay. Will also attempt writing the relevant tests.

The `if (index === 'new')` statement is passed from the frontend when a user saves a new section. It was just so that we could distinguish between new sections and existing ones. This is probably going to change when issue #80 is addressed.

Yeah, the frontend code was pretty long. We refactored the addNewSection function and the html for the about page. See pull request #78 (sorry there's still a lot of commented out code in there)
